### ISSUE TITLE
I can't stop the protocol on Opentrons

### ISSUE THREAD

**Issue by noparis** [FlorianBauer/ot2-controller#4]
Hello,

I am using ot2-controller as server and sila_unniversal_client.
When I run the protocol, I can't stop the Opentrons. Even if I delete the .py script and stop the server.
To stop the execution, I have to turn off the Opentrons.

Thanks for you help.

**Comment by FlorianBauer** [comment]
Hi,

without knowing what's exactly going on in the protocol, which commands are invoked or any other detailed information, it is not possible for me to troubleshoot the issue. You can try to run the protocol with a different client (e.g. the latest release of the [sila-orchestrator](https://github.com/FlorianBauer/sila-orchestrator/releases) or try to put some print statements inside the protocol (or even the ot2-controller) to find the causing command for this behavior. Also, a look into the log-files of the ot2 might bring more insight.

If you were able to successfully narrow down the issue, it would be great if you can share your findings here, so others with the same problem might learn from your experience.

Best regards and good luck.

**Comment by FlorianBauer** [comment]
> If I want to pause in the middle of the protocol and then restart it, or just stop it, I can't.

Yes, this behavior is correct and expected. This is due to the way the ot2-controller interacts with the device, since the controller basically just uploads the given `*.py`-protocol onto the device memory and tells the OT-2 to execute it. After that, the responsibility lays solely on the OT-2. Its kind of a fire and forget operation for the corresponding SiLA client.

A crude workaround to this limitation would be to slice a protocol into smaller `*.py`-protocols and execute them step by step in the SiLA client of your choice.

> I don't use SiLA-orchestrator, because I can't import the json files even if I put .silo.

The `*.silo`-file contains only internal information dedicated to the sila-orchestrator alone. This file format shall not be used in combination with other applications and vice versa.

Best regards.
